core/src/test/scala/akka/kafka/scaladsl/IntegrationSpec.scala

```diff
@@ -197,9 +208,70 @@ class IntegrationSpec extends TestKit(ActorSystem("IntegrationSpec"))
       Await.result(control.isShutdown, remainingOrDefault)
     }
 
-    // TODO add more tests
+    "consume and commit in batches" in {
+      givenInitializedTopic()
 
-  }
+      produceMessages(topic1, 1 to 100)
+
+      val consumerSettings = createConsumerSettings(topic1, group1, client1)
 
+      def consumeAndBatchCommit() = {
+        Consumer.committableSource(consumerSettings)
+          .map { msg => { ; msg.committableOffset } }
+          .batch(max = 10, first => CommittableOffsetBatch.empty.updated(first)) { (batch, elem) =>
+            batch.updated(elem)
+          }
+          .mapAsync(1)(_.commitScaladsl()).toMat(TestSink.probe)(Keep.both)
```

Comment:
Upon review there's a more critical issue here.  ```run()``` is not being called.  I also have a question.  For verification using a probe, it seems like we cannot expect to verify based on the number of messages produced to the topic (because of batching).  What kind of test verification can we do in this case ?